plugins/system/userlogs/userlogs.php

```diff
@@ -46,78 +74,92 @@ public function __construct(&$subject, $config)
-         $dispatcher = JEventDispatcher::getInstance();
-         $query = $this->db->getQuery(​true​);
-         if($this->params->get('ip_logging'))
-         {
-             $this->jinput = JFactory::getApplication()->input;
-             $ip_address = $this->jinput->server->get('REMOTE_ADDR');
-         }
-         else
-         {
-             $ip_address = JText::_('PLG_SYSTEM_USERLOG_DISABLED');
-         }
-         $columns = array('message', 'log_date', 'extension', 'user_id', 'ip_address');
-         $values = array($this->db->quote($message), $this->db->quote($date), $this->db->quote($context), $this->db->quote($user->id), $this->db->quote($ip_address));
-         $query
-             ->insert($this->db->quoteName('#__user_logs'))
-             ->columns($this->db->quoteName($columns))
-             ->values(implode(',', $values));
-         $this->db->setQuery($query);
-         $this->db->execute();
-         $dispatcher->trigger('onUserLogsAfterMessageLog', array ($message, $date, $context, $user->name, $ip_address));
-     }
-
-     /**
- 	 * Function to check if a component is loggable or not
- 	 *
- 	 * @param   string   $extension  The extension that triggered the event
- 	 *
- 	 * @return  boolean
- 	 *
- 	 * @since   3.7
- 	 */
-     protected function checkLoggable($extension)
-     {
-         if(!in_array($extension, $this->loggable_extensions))
-         {
-             return false;
-         }
-
-         return true;
-     }
-
-    /**
+	protected function checkLoggable($extension)
+	{
+		if (!in_array($extension, $this->loggable_extensions))
+		{
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
 	 * After save content logging method
 	 * This method adds a record to #__user_logs contains (message, date, context, user)
 	 * Method is called right after the content is saved
 	 *
 	 * @param   string   $context  The context of the content passed to the plugin
 	 * @param   object   $article  A JTableContent object
-	 * @param   boolean  $isNew    If the content is just about to be created
+	 * @param   boolean  $isNew	If the content is just about to be created
 	 *
 	 * @return  boolean   true if function not enabled, is in front-end or is new. Else true or
-	 *                    false depending on success of save function.
+	 *					false depending on success of save function.
 	 *
 	 * @since   3.7
 	 */
-    public function onContentAfterSave($context, $article, $isNew)
-    {
-        if ($this->checkLoggable($this->jinput->get('option')))
-        {
-            $isNew_string = $isNew ? 'true' : 'false';
-            $message = '{"title":"'.$article->title.'","isNew":"'.$isNew_string.'", "event":"onContentAfterSave", "id":'.$article->id.'}';
-            $strContext = (string)$context;
-            $this->addLogsToDB($message, $strContext);
-        }
-    }
-
-    /**
+	public function onContentAfterSave($context, $content, $isNew)
+	{
+		if ($this->checkLoggable($this->app->input->get('option')))
```

Comment:
change that to:

if (!$this->checkLoggable($this->app->input->get('option')))
{
    return;
}

Then you don't have to surround that whole block with ifs